src/index.js

```diff
@@ -1,98 +1,105 @@
-import {List, Map} from 'immutable';
+import { findIndex } from './array-utils';
 
 export const BEGIN = '@@optimist/BEGIN';
 export const COMMIT = '@@optimist/COMMIT';
 export const REVERT = '@@optimist/REVERT';
 
 export const ensureState = state => {
-  if (Map.isMap(state)) {
-    if (List.isList(state.get('history'))) {
-      return state.get('current');
+  if (state instanceof Object) {
```

Comment:
we can get rid of this first `state instanceof Object` since state.history ensures that it is an Object.